### PR TITLE
RSDK-2273 Lock model during Read func

### PR DESCRIPTION
**Pull request by kim-mishra**
When one model was being used for two cameras, the model was being invoked by both cameras at the same time which caused a seg fault. By locking the model during an individual call to `Read`, both cameras are able access the same instance of a model.

### CODE REVIEW

**Review by bhaney — CHANGES_REQUESTED** [review]
Thanks for this! Could you add a test  in `services/vision/builtin/tflite_test.go` that sets up a classifier, and then tries to call it twice in two separate go routines, to try to simulate a simultaneous call to GetClassifications?

Also, did you test the bug situation in the ticket, and did it no longer crash?

File: services/vision/builtin/tflite_classifier.go [hunk 1]
```diff
@@ -76,6 +76,9 @@ func NewTFLiteClassifier(ctx context.Context, conf *vision.VisModelConfig,
 
 	// This function that gets returned should be the Classifier
 	return func(ctx context.Context, img image.Image) (classification.Classifications, error) {
```
Comment (bhaney) [inline services/vision/builtin/tflite_classifier.go:78]: could you also add this to the Detections function in tflite_detector?

**Comment by kim-mishra** [comment]
> Thanks for this! Could you add a test in `services/vision/builtin/tflite_test.go` that sets up a classifier, and then tries to call it twice in two separate go routines, to try to simulate a simultaneous call to GetClassifications?
> 
> Also, did you test the bug situation in the ticket, and did it no longer crash?

Yup! Added in the tests and testing the situation on the ticket manually (and it did not crash)

**Review by bhaney — APPROVED** [review]
LGTM! thanks!

File: ml/inference/tflite.go [hunk 2]
```diff
@@ -123,6 +125,7 @@ func (loader TFLiteModelLoader) Load(modelPath string) (*TFLiteStruct, error) {
 		interpreterOptions: loader.interpreterOptions,
 		Info:               info,
 		modelPath:          modelPath,
+		Mu:                 sync.Mutex{},
```
Comment (edaniels) [inline ml/inference/tflite.go:128]: should be unexported

**Review by edaniels — CHANGES_REQUESTED** [review]

File: ml/inference/tflite.go (hunk 2 above)
Comment (kim-mishra) [reply ml/inference/tflite.go:128]: This is the inference package and I am locking it in the builtin package, so it needs to be exported. Should I lock it somewhere else?

**Review by edaniels — APPROVED** [review]

**Pull request merged**